core/commands/dag/dag.go

```diff
@@ -241,3 +277,487 @@ var DagResolveCmd = &cmds.Command{
 	},
 	Type: ResolveOutput{},
 }
+
+// It is not guaranteed that a root in a header is actually present in the same ( or any )
+// .car file. This is the case in version 1, and ideally in further versions too
+// Accumulate any root CID seen in a header, and supplement its actual node if/when encountered
+// We will attempt a pin *only* at the end in case all car files were well formed
+//
+// The boolean value indicates whether we have encountered the root within the car file's
+type expectedRootsSeen map[cid.Cid]bool
+type importResult struct {
+	roots *expectedRootsSeen
```

Comment:
Maps are already allocated on the heap. We don't need the pointer.

Reply:
Fixed in https://github.com/ipfs/go-ipfs/pull/7038